app/code/Magento/Catalog/Model/Product/Type/AbstractType.php

```diff
@@ -603,7 +603,7 @@ protected function _prepareOptions(\Magento\Framework\DataObject $buyRequest, $p
                 }
             }
             if (count($results) > 0) {
-                throw new LocalizedException(__(implode("\n", $results)));
+                throw new LocalizedException(__(implode("\n", array_unique($results))));
```

Comment:
https://github.com/magento/graphql-ce/blob/1f8c1f301f0e3083bc5d64cbdde021876995e381/app/code/Magento/Catalog/Model/Product/Type/AbstractType.php#L596

Reply:
Hey @naydav , 

I see what do you mean, however on Magento storefront current behaviour is done on presentation layer in `app/code/Magento/Checkout/Controller/Cart/Add.php:155` - and yes it is some kind of a duck-tape :)
But still, I do suggest to do the same - create the same "duck-tape" on the same layer (presentation) here `app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php:84` - in GraphQl.
And create a separate issue in Magento core to solve the problem with multiple messages from `app/code/Magento/Catalog/Model/Product/Type/AbstractType.php` (domain layer).

Moreover, we are not able to make fixes of Magento core in GraphQl project according to this [article](https://github.com/magento/graphql-ce/wiki/Approach-to-Implementation#technical-requirements-to-features), so the current solution will be the best I suppose.

@naydav - What do you think about it?
Thx!

Reply:
@paliarush - could you check my prev comment?
Thx!

Reply:
> Hey @naydav ,
> 
> I see what do you mean, however on Magento storefront current behaviour is done on _presentation_ layer in `app/code/Magento/Checkout/Controller/Cart/Add.php:155` - and yes it is some kind of a duck-tape :)
> 
> But still, I do suggest to do the same - create the same "duck-tape" on the same _presentation_ layer here `app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php:84` - in GraphQl.
> And create a separate issue in Magento core to solve the problem with multiple messages from `app/code/Magento/Catalog/Model/Product/Type/AbstractType.php` (_domain_ layer).
> 
> Moreover, we are not able to make fixes of Magento core in GraphQl project according to this [article](https://github.com/magento/graphql-ce/wiki/Approach-to-Implementation#technical-requirements-to-features), so the current solution will be the best I suppose.
> 
> @naydav - What do you think about it?
> Thx!

Agree. Because the current fix is BIC.
Thanks for the review.